.github/workflows/on_pull_request_synchronized.yml

```diff
@@ -0,0 +1,44 @@
+name: Pull request synchronized
+on:
+  pull_request:
+    types:
+      - synchronize
+jobs:
+  disable-automerge:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v7
+        with:
+          script: |
+            const prQuery = `query PullRequest($owner: String!, $repo: String!, $pullRequestNumber: Int!) {
+              repository(owner: $owner, name: $repo) {
+                pullRequest(number: $pullRequestNumber) {
+                  id
+                  autoMergeRequest {
+                    enabledAt
+                  }
+                }
+              }
+            }`;
+            const prVariables = {
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pullRequestNumber: context.issue.number
+            }
+            const prResult = await github.graphql(prQuery, prVariables)
+            if (!prResult.repository.pullRequest.autoMergeRequest) {
+              console.log('Auto merge is not enabled')
+              return
+            }
+            const automergeQuery = `mutation DisablePullRequestAutoMerge($pullRequestId: ID!) {
+              disablePullRequestAutoMerge(input: {pullRequestId: $pullRequestId}) {
+                pullRequest {
+                  id
+                }
+              }
+            }`;
```

Comment:
will this have a security issue? like can I now add a github workflows and disable other PR's automerge button?

what kind of credential is this github actions using?

Reply:
i didn't create any credentials for this action; the github bot has write access; i think anyone who has the 'write' access can create a github workflow to disable other PR's automerge button yes, but since they have 'write' access, their credential can perform that action so they can basically write a local script to do so

i don't think the github bot can enable automerge since that is protected by branch protection; let's confirm next week